Sources/ClientRuntime/Networking/Http/HTTPResponseDocumentBinding.swift

```diff
@@ -0,0 +1,19 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import class SmithyXML.Reader
+
+public typealias HTTPResponseDocumentBinding<Reader> = (HttpResponse) async throws -> Reader
+
+/// Creates a `HTTPResponseDocumentBinding` for converting a HTTP response into a `Reader`.
+public var responseDocumentBinding: HTTPResponseDocumentBinding<Reader> {
```

Comment:
Creates a Reader from response data, which can then be used to deserialize the response.